webapi/src/main/scala/org/knora/webapi/store/triplestore/impl/TriplestoreServiceLive.scala

```diff
@@ -437,226 +393,103 @@ case class TriplestoreServiceLive(
+                .header("Accept", mimeTypeApplicationNQuads),
+            )
+          body <- ensuringBody(response.body)
+          _    <- ZIO.attempt(Files.write(outputFile, body.getBytes(StandardCharsets.UTF_8), CREATE, TRUNCATE_EXISTING))
+        } yield ()
       case MigrateSpecificGraphs(graphIris) =>
-        ZIO.foreach(graphIris)(graphIri => downloadGraph(graphIri, zio.nio.file.Path.fromJava(outputFile), NQuads)).unit
+        ZIO.foreach(graphIris)(downloadGraph(_, NioPath.fromJava(outputFile), NQuads)).unit
     }
-  }
+
+  private def ensuringBody(body: Either[String, String]): Task[String] =
+    ZIO
+      .succeed(body.toOption)
+      .someOrFail(TriplestoreResponseException("Triplestore returned no content for for repository dump"))
 
   /**
    * Uploads repository content from an N-Quads file.
    *
    * @param inputFile an N-Quads file containing the content to be uploaded to the repository.
    */
-  override def uploadRepository(inputFile: Path): Task[Unit] = {
-    val fileEntity        = new FileEntity(inputFile.toFile, ContentType.create(mimeTypeApplicationNQuads, "UTF-8"))
-    val request: HttpPost = new HttpPost(paths.repository)
-    request.setEntity(fileEntity)
-    doHttpRequest(request, _ => ZIO.unit).unit
-  }
-
-  /**
-   * Makes an HTTP connection to the triplestore, and delegates processing of the response
-   * to a function.
-   *
-   * @param request         the request to be sent.
-   * @tparam T the return type of `processError`.
-   * @return the return value of `processError`.
-   */
-  private def doHttpRequest[T](
-    request: HttpRequest,
-    processResponse: CloseableHttpResponse => Task[T],
-  ): Task[T] = {
-
-    def executeQuery(): Task[CloseableHttpResponse] = {
-      ZIO
-        .attempt(queryHttpClient.execute(targetHost, request))
-        .catchSome {
-          case socketTimeoutException: java.net.SocketTimeoutException =>
-            val message =
-              "The triplestore took too long to process a request. This can happen because the triplestore needed too much time to search through the data that is currently in the triplestore. Query optimisation may help."
-            val error = TriplestoreTimeoutException(message, socketTimeoutException)
-            ZIO.logError(error.toString) *>
-              ZIO.fail(error)
-          case e: Exception =>
-            val message = s"Failed to connect to triplestore."
-            val error   = TriplestoreConnectionException(message, Some(e))
-            ZIO.logError(error.toString) *>
-              ZIO.fail(error)
-        }
+  override def uploadRepository(inputFile: Path): Task[Unit] =
+    ZIO
+      .readFile(inputFile)
+      .map(
+        authenticatedRequest
+          .post(targetHostUri.addPath(paths.repository))
+          .contentType(mimeTypeApplicationNQuads, "UTF-8")
+          .body(_),
+      )
+      .flatMap(doHttpRequestSttp)
+      .unit
+
+  private def doHttpRequestSttp[T](
```

Comment:
nitpick: I would prefer keeping the method name `doHttpRequest`, adding `STTP` doesn't really add any value

Reply:
An artifact of the refactoring process, renaming.